server/src/main/java/org/elasticsearch/common/util/set/Sets.java

```diff
@@ -104,35 +96,13 @@ public static <T> Set<T> difference(Set<T> left, Set<T> right) {
      * @return the sorted relative complement of the left set with respect to the right set
      */
     public static <T> SortedSet<T> sortedDifference(final Set<T> left, final Set<T> right) {
```

Comment:
Is it worth changing this (and others) to `NavigableSet` whilst we're here?

Reply:
Or having a general `<S extends Set<T>, T> S difference(Supplier<S>, Set<T>, Set<T>)` method so you can specify your own set implementation?

Reply:
We're only using this in a single spot to begin with :) No need to generalize it I think.